### PR TITLE
Java script

### PR DESCRIPTION
**Pull request by ayushiagrawal2000**
JS Dynamic calendar

### CODE REVIEW

File: Assignment 3-JS/sass/style.scss [hunk 1]
```diff
@@ -0,0 +1,235 @@
+body {
+  margin: 0;
+  padding: 0;
+  font-family: sans-serif;
+}
+html {
+  font-size: 80%;
```
Comment (pranav96vaidya) [inline Assignment 3-JS/sass/style.scss:7]: No need.

File: Assignment 3-JS/sass/style.scss [hunk 2]
```diff
@@ -0,0 +1,235 @@
+body {
+  margin: 0;
+  padding: 0;
```
Comment (pranav96vaidya) [inline Assignment 3-JS/sass/style.scss:3]: Check for its usage.

File: Assignment 3-JS/sass/style.scss [hunk 3]
```diff
@@ -0,0 +1,235 @@
...
+  font-family: sans-serif;
+}
+html {
+  font-size: 80%;
+}
+.container-body {
```
Comment (pranav96vaidya) [inline Assignment 3-JS/sass/style.scss:9]: Use nested code

File: Assignment 3-JS/sass/style.scss [hunk 4]
```diff
@@ -0,0 +1,235 @@
...
+  width: 100%;
```
Comment (pranav96vaidya) [inline Assignment 3-JS/sass/style.scss:10]: Block level element takes default 100% width.

File: Assignment 3-JS/sass/style.scss [hunk 5]
```diff
@@ -0,0 +1,235 @@
...
+  height: 100%;
+  background:linear-gradient(to left,rgb(163, 243, 226),rgb(45, 190, 190),rgb(2, 129, 129));
+  color: white;
```
Comment (pranav96vaidya) [inline Assignment 3-JS/sass/style.scss:13]: Move into a new file and use that here

File: Assignment 3-JS/sass/style.scss [hunk 6]
```diff
@@ -0,0 +1,235 @@
...
+  .calender {
+      width: 460px;
+      background-color: #00001a;   
+      box-shadow: 5px 5px 10px black;
+      .month {
+          height: 20%;
+          display: flex;
+          justify-content: space-between;
+          align-items: center;
+          text-align: center;
```
Comment (pranav96vaidya) [inline Assignment 3-JS/sass/style.scss:26]: Check its usage.

File: Assignment 3-JS/sass/style.scss [hunk 7]
```diff
@@ -0,0 +1,235 @@
...
+          padding: 0 20px;
+          border-bottom:2px solid #00e6b8;
+      }
+      .month i {
+          font-size: 20px;
+          cursor: pointer;
+          color: #00e6b8;
+      }
+      .month .date .month-name, .current-year-name {
```
Comment (pranav96vaidya) [inline Assignment 3-JS/sass/style.scss:35]: Fix it

File: Assignment 3-JS/sass/style.scss [hunk 8]
```diff
@@ -0,0 +1,235 @@
...
+          text-transform: uppercase;
+          letter-spacing: 2px;
+          font-size: 30px;
+          color: #00e6b8;
+          cursor: pointer;
+      }
+      .week-days {
+          display: flex;
+          padding: 20px 0px;
+
```
Comment (pranav96vaidya) [inline Assignment 3-JS/sass/style.scss:45]: Remove extra lines

File: Assignment 3-JS/sass/style.scss [hunk 9]
```diff
@@ -0,0 +1,235 @@
...
+          cursor: pointer;
+      }
+      .week-days {
+          display: flex;
+          padding: 20px 0px;
+          
+      }
+      .week-days div {
+          width: calc(460px/8);
+          display: flex;
```
Comment (pranav96vaidya) [inline Assignment 3-JS/sass/style.scss:49]: Remove flex property.

File: Assignment 3-JS/sass/style.scss [hunk 10]
```diff
@@ -0,0 +1,235 @@
...
+          justify-content: center;
+          align-items: center;
+      }
+      .days-container {
+          display:flex;
+            #week-number{
```
Comment (pranav96vaidya) [inline Assignment 3-JS/sass/style.scss:55]: Add a space

File: Assignment 3-JS/sass/style.scss [hunk 11]
```diff
@@ -0,0 +1,235 @@
...
+                  box-sizing:border-box;
+              }
+          }
+      }
+      .days {
+          display: flex;
+          flex-wrap: wrap;
+          width:403.3px;
+      }
+      .days div {
```
Comment (pranav96vaidya) [inline Assignment 3-JS/sass/style.scss:72]: use a class

File: Assignment 3-JS/sass/style.scss [hunk 12]
```diff
@@ -0,0 +1,235 @@
...
+      .today {
+          border:2px solid #00e6b8;
+          box-sizing: border-box;
+      }
+      .days div:hover:not(.week-number):not(.event){
+          box-sizing: border-box;
+          border: 2px solid white;
+      }
+  }
+  .sidepanel {
```
Comment (pranav96vaidya) [inline Assignment 3-JS/sass/style.scss:101]: rename it.

File: Assignment 3-JS/sass/style.scss [hunk 13]
```diff
@@ -0,0 +1,235 @@
...
+      color: #818181;
+      transition: 0.3s;
+      cursor: pointer;
+      height: 115px;
+      display: flex;
+      flex-wrap: wrap;
+      align-items: center;
+      box-sizing: border-box;
+  }
+  .sidepanel #icon1 .next {
```
Comment (pranav96vaidya) [inline Assignment 3-JS/sass/style.scss:129]: Is it useful?

File: Assignment 3-JS/sass/style.scss [hunk 14]
```diff
@@ -0,0 +1,235 @@
...
+      margin: 0;
+      padding: 0;
+  }
+  .sidepanel #icon1,#icon2 {
+      font-size:20px;
```
Comment (pranav96vaidya) [inline Assignment 3-JS/sass/style.scss:134]: Use it in a single place.

File: Assignment 3-JS/sass/style.scss [hunk 15]
```diff
@@ -0,0 +1,235 @@
...
+              line-height:30px;
+          }
+      }
+      #modal-title {
+          font-size: 20px;
+      }
+    }
+  }
+}
+@media screen and (max-width:700px){
```
Comment (pranav96vaidya) [inline Assignment 3-JS/sass/style.scss:174]: use inline media query

File: Assignment 3-JS/sass/style.scss [hunk 16]
```diff
@@ -0,0 +1,235 @@
...
+        height: 160px;
+      }
+      .sidepanel #years .year-name {
+        width: calc(460px/5);
+        height: 80px;
+        font-size: 25px;
+      }  
+    }
+  }
+}
```
Comment (pranav96vaidya) [inline Assignment 3-JS/sass/style.scss:235]: Add a new line at the end of file.

File: Assignment 3-JS/script/script.js [hunk 17]
```diff
@@ -0,0 +1,320 @@
+
```
Comment (pranav96vaidya) [inline Assignment 3-JS/script/script.js:1]: Add document ready function and move everything inside that function.

File: Assignment 3-JS/script/script.js [hunk 18]
```diff
@@ -0,0 +1,320 @@
...
+      else {
+          weeknum = Math.floor((daynum+day-1)/7);
+      }
+      return weeknum;
+  };
+  var title = document.getElementById('modal-title');
+  var body = document.querySelector('.modal-body p');
+  var event_date = document.getElementById('event-date');
+  function triggerModal(obj){
+    console.log(obj);
```
Comment (pranav96vaidya) [inline Assignment 3-JS/script/script.js:32]: remove console

File: Assignment 3-JS/script/script.js [hunk 19]
```diff
@@ -0,0 +1,320 @@
...
+          yearElement.classList.add('firstElement');
+          yearElement.style.color = " #00e6b8 ";
+        }
+      }
+  }
+
+let arrow_up = document.getElementById('icon1');
+
+arrow_up.addEventListener('click',function(){
+  var Element = document.querySelector('.firstElement');
```
Comment (pranav96vaidya) [inline Assignment 3-JS/script/script.js:74]: Revisit all the variable's name

File: Assignment 3-JS/script/script.js [hunk 20]
```diff
@@ -0,0 +1,320 @@
...
+    }else{
+      renderCalendar(monthIndex-1,parseInt(yearcheck));
+    }
+});
+  function nextMonth(){
+    var monthcheck = document.querySelector('.month-name').innerHTML;
+    var yearcheck = document.querySelector('.current-year-name').innerHTML;
+    
+    var monthIndex = monthArray.indexOf(monthcheck.toLowerCase());
+    if(monthIndex == 11){
```
Comment (pranav96vaidya) [inline Assignment 3-JS/script/script.js:312]: Add space between if and "("

File: Assignment 3-JS/template/index.html [hunk 21]
```diff
@@ -0,0 +1,76 @@
+<html lang="en">
+
+<head>
+    <meta charset="UTF-8">
+    <meta http-equiv="X-UA-Compatible" content="IE=edge">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <title>Document</title>
+    <link rel="stylesheet" href="../style/style.css">
+    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.13.0/css/all.min.css" />
+    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/animated.css/4.1.1/animated.min.css" />
```
Comment (pranav96vaidya) [inline Assignment 3-JS/template/index.html:10]: remove it

File: Assignment 3-JS/template/index.html [hunk 22]
```diff
@@ -0,0 +1,76 @@
+<html lang="en">
+
+<head>
+    <meta charset="UTF-8">
+    <meta http-equiv="X-UA-Compatible" content="IE=edge">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <title>Document</title>
```
Comment (pranav96vaidya) [inline Assignment 3-JS/template/index.html:7]: Rename it to "Calendar"

File: Assignment 3-JS/template/index.html [hunk 23]
```diff
@@ -0,0 +1,76 @@
...
+                <div>Mon</div>
+                <div>Tue</div>
+                <div>Wed</div>
+                <div>Thu</div>
+                <div>fri</div>
+                <div>Sat</div>
+            </div>
+            <div class="days-container" onclick="closeNav()">
+                <div id="week-number">
+                    <br>
```
Comment (pranav96vaidya) [inline Assignment 3-JS/template/index.html:47]: remove line break

**Review by pranav96vaidya — COMMENTED** [review]
Please fix these ASAP

File: Assignment 3-JS/sass/style.scss (hunk 1 above)
Comment (ayushiagrawal2000) [reply Assignment 3-JS/sass/style.scss:7]: removed it.

File: Assignment 3-JS/sass/style.scss (hunk 2 above)
Comment (ayushiagrawal2000) [reply Assignment 3-JS/sass/style.scss:3]: No need for this so removed it.

File: Assignment 3-JS/sass/style.scss (hunk 3 above)
Comment (ayushiagrawal2000) [reply Assignment 3-JS/sass/style.scss:9]: Placed it inside body and made code nested.

File: Assignment 3-JS/sass/style.scss (hunk 4 above)
Comment (ayushiagrawal2000) [reply Assignment 3-JS/sass/style.scss:10]: Removed width of block element.

File: Assignment 3-JS/sass/style.scss (hunk 5 above)
Comment (ayushiagrawal2000) [reply Assignment 3-JS/sass/style.scss:13]: Moved it in another sass file and imported it here.

File: Assignment 3-JS/sass/style.scss (hunk 6 above)
Comment (ayushiagrawal2000) [reply Assignment 3-JS/sass/style.scss:26]: It is used to place month name, year and right-left arrows in a single line, and place them in equal space and vertically in center of the container.

File: Assignment 3-JS/sass/style.scss (hunk 7 above)
Comment (ayushiagrawal2000) [reply Assignment 3-JS/sass/style.scss:35]: Fixed it.

File: Assignment 3-JS/sass/style.scss (hunk 8 above)
Comment (ayushiagrawal2000) [reply Assignment 3-JS/sass/style.scss:45]: Removed it.

File: Assignment 3-JS/sass/style.scss (hunk 9 above)
Comment (ayushiagrawal2000) [reply Assignment 3-JS/sass/style.scss:49]: Removed it.

File: Assignment 3-JS/sass/style.scss (hunk 10 above)
Comment (ayushiagrawal2000) [reply Assignment 3-JS/sass/style.scss:55]: Done.

File: Assignment 3-JS/sass/style.scss (hunk 11 above)
Comment (ayushiagrawal2000) [reply Assignment 3-JS/sass/style.scss:72]: Gave a common class to all div elements inside it's wrapper.

File: Assignment 3-JS/sass/style.scss (hunk 12 above)
Comment (ayushiagrawal2000) [reply Assignment 3-JS/sass/style.scss:101]: Renamed it to Scrollpanel.

File: Assignment 3-JS/sass/style.scss (hunk 13 above)
Comment (ayushiagrawal2000) [reply Assignment 3-JS/sass/style.scss:129]: No it is not, so removed it.

File: Assignment 3-JS/sass/style.scss (hunk 14 above)
Comment (ayushiagrawal2000) [reply Assignment 3-JS/sass/style.scss:134]: Made a sass variable named $font-size and using it.

File: Assignment 3-JS/sass/style.scss (hunk 15 above)
Comment (ayushiagrawal2000) [reply Assignment 3-JS/sass/style.scss:174]: Done.

File: Assignment 3-JS/sass/style.scss (hunk 16 above)
Comment (ayushiagrawal2000) [reply Assignment 3-JS/sass/style.scss:235]: Done.

File: Assignment 3-JS/script/script.js (hunk 17 above)
Comment (ayushiagrawal2000) [reply Assignment 3-JS/script/script.js:1]: Moved everything inside document ready function except onclick functions.

File: Assignment 3-JS/script/script.js (hunk 18 above)
Comment (ayushiagrawal2000) [reply Assignment 3-JS/script/script.js:32]: Removed console from everywhere in the code.

File: Assignment 3-JS/script/script.js (hunk 19 above)
Comment (ayushiagrawal2000) [reply Assignment 3-JS/script/script.js:74]: Changed variable names to more readable names.

File: Assignment 3-JS/script/script.js (hunk 20 above)
Comment (ayushiagrawal2000) [reply Assignment 3-JS/script/script.js:312]: Done.

File: Assignment 3-JS/template/index.html (hunk 21 above)
Comment (ayushiagrawal2000) [reply Assignment 3-JS/template/index.html:10]: Removed it.

File: Assignment 3-JS/template/index.html (hunk 22 above)
Comment (ayushiagrawal2000) [reply Assignment 3-JS/template/index.html:7]: Done.

File: Assignment 3-JS/template/index.html (hunk 23 above)
Comment (ayushiagrawal2000) [reply Assignment 3-JS/template/index.html:47]: Done.

File: Assignment 3-JS/script/script.js [hunk 24]
```diff
@@ -0,0 +1,312 @@
+  //to render years
+  function Years(){
```
Comment (pranav96vaidya) [inline Assignment 3-JS/script/script.js:2]: Add a space after ()

File: Assignment 3-JS/script/script.js [hunk 25]
```diff
@@ -0,0 +1,312 @@
...
+        yearElement.scrollIntoView();
+        yearElement.classList.add('firstElement-to-scroll');
+        yearElement.style.color = " #00e6b8 ";
+      }
+    }
+  }
+  //to hide the side-panel
+  function closeNav() {
+    document.getElementById("myScrollpanel").style.height = "0";
+  }
```
Comment (pranav96vaidya) [inline Assignment 3-JS/script/script.js:25]: Indentation is not proper and add a line break.

File: Assignment 3-JS/sass/style.scss [hunk 26]
```diff
@@ -0,0 +1,183 @@
...
+      .today {
+        border: 2px solid $aqua-green;
+        box-sizing: border-box;
+      }
+      .days .number-of-days:hover:not(.week-number):not(.event){
+        box-sizing: border-box;
+        border: 2px solid $white;
+      }
+    }
+    .Scrollpanel {
```
Comment (pranav96vaidya) [inline Assignment 3-JS/sass/style.scss:109]: Class name should start with lower case and it should be hyphen separated if more than 1 word is used.

File: Assignment 3-JS/script/script.js [hunk 27]
```diff
@@ -0,0 +1,312 @@
...
+      weeknum = Math.floor((daynum+day-1)/7);
+    }
+    return weeknum;
+  };
+  
+    var title = document.getElementById('modal-title');
+    var body = document.querySelector('.modal-body p');
+    var event_date = document.getElementById('event-date');
+    function triggerModal(obj) {
+      console.log(obj);
```
Comment (pranav96vaidya) [inline Assignment 3-JS/script/script.js:54]: remove console

File: Assignment 3-JS/script/script.js [hunk 28]
```diff
@@ -0,0 +1,312 @@
...
+      title.innerHTML = obj.title;
+      body.innerHTML = obj.discription;
+      event_date.innerHTML = obj.date;
+      $('#myModal').modal('show');
+    }
+  
+  //to open side-panel
+  document.getElementsByClassName('current-year-name')[0].addEventListener('click', () => {
```
Comment (pranav96vaidya) [inline Assignment 3-JS/script/script.js:62]: Check for indentation

File: Assignment 3-JS/script/script.js [hunk 29]
```diff
@@ -0,0 +1,312 @@
...
+        if (value > 2050) {
+          var scroll = year_names[0];
+          scroll.scrollIntoView();
+        }
+      }
+  });
+  
+  let arrow_down = document.getElementById('icon2');
+  
+  arrow_down.addEventListener('click',function(){
```
Comment (pranav96vaidya) [inline Assignment 3-JS/script/script.js:90]: Check for spacing in whole file

File: Assignment 3-JS/script/script.js [hunk 30]
```diff
@@ -0,0 +1,312 @@
...
+    value = value - 10;
+    var year_names = document.getElementsByClassName('year-name');
+      for(let i = 0; i < year_names.length; i++) {
+        var check = year_names[i].innerHTML;
+        if(check == value){
+          scrollElement.classList.remove('firstElement-to-scroll');
+          year_names[i].classList.add('firstElement-to-scroll');
+          let nextScroll = document.querySelector('.firstElement-to-scroll');
+          nextScroll.scrollIntoView();
+        } if (value < 1950) {
```
Comment (pranav96vaidya) [inline Assignment 3-JS/script/script.js:102]: if should be in next line.

File: Assignment 3-JS/script/script.js [hunk 31]
```diff
@@ -0,0 +1,312 @@
...
+          "august",
+          "september",
+          "october",
+          "november",
+          "december"
+      ];
+  
+  var events = {
+    "january-1": {
+      title: " Happy New year",
```
Comment (pranav96vaidya) [inline Assignment 3-JS/script/script.js:132]: What is the need of a space at the start of value?

File: Assignment 3-JS/script/script.js [hunk 32]
```diff
@@ -0,0 +1,312 @@
...
+      nature to survive; in this century he is beginning to 
+      realise that, in order to survive, he must protect it.`
+    }
+  }
+   
+  const renderCalendar = (month,year) => {
+   
+    document.getElementsByClassName('month-name')[0].innerHTML = monthArray[month];
+    document.getElementsByClassName('current-year-name')[0].innerHTML = year;
+
```
Comment (pranav96vaidya) [inline Assignment 3-JS/script/script.js:181]: Why so many line breaks?

File: Assignment 3-JS/script/script.js [hunk 33]
```diff
@@ -0,0 +1,312 @@
...
+    var weekOfEndDate = new Date(dateinput2.getFullYear(),dateinput2.getMonth(),dateinput2.getDate());
+    var endDate = weekOfEndDate.getWeek();
+  
+    let week = "";
+    for(let j = startDate; j <= endDate; j++) { 
+      week += `<div class="week">${j}</div>`;
+    }
+    weekDays.innerHTML = week;
+  
+    month_days.innerHTML = "";
```
Comment (pranav96vaidya) [inline Assignment 3-JS/script/script.js:207]: Use variable name in camel-case format.

File: Assignment 3-JS/template/index.html [hunk 34]
```diff
@@ -0,0 +1,65 @@
...
+          <div class="week-day-name">Sat</div>
+      </div>
+      <div class="days-container" onclick="closeNav()">
+        <div id="week-number">
+        </div>
+        <div class="days">
+        </div>
+      </div>
+    </div>
+      <div id="myModal" class="modal fade " role="dialog">
```
Comment (pranav96vaidya) [inline Assignment 3-JS/template/index.html:49]: Fix indentation here

**Review by pranav96vaidya — COMMENTED** [review]
Fix these comments too.

File: Assignment 3-JS/script/script.js (hunk 24 above)
Comment (ayushiagrawal2000) [reply Assignment 3-JS/script/script.js]: Done.

File: Assignment 3-JS/script/script.js (hunk 25 above)
Comment (ayushiagrawal2000) [reply Assignment 3-JS/script/script.js]: Done.

File: Assignment 3-JS/sass/style.scss (hunk 26 above)
Comment (ayushiagrawal2000) [reply Assignment 3-JS/sass/style.scss]: Changed it to scroll-panel.

File: Assignment 3-JS/script/script.js (hunk 27 above)
Comment (ayushiagrawal2000) [reply Assignment 3-JS/script/script.js]: Removed it.

File: Assignment 3-JS/script/script.js (hunk 28 above)
Comment (ayushiagrawal2000) [reply Assignment 3-JS/script/script.js:60]: Fixed it.

File: Assignment 3-JS/script/script.js (hunk 29 above)
Comment (ayushiagrawal2000) [reply Assignment 3-JS/script/script.js]: Fixed it

File: Assignment 3-JS/script/script.js (hunk 30 above)
Comment (ayushiagrawal2000) [reply Assignment 3-JS/script/script.js]: Done.

File: Assignment 3-JS/script/script.js (hunk 31 above)
Comment (ayushiagrawal2000) [reply Assignment 3-JS/script/script.js]: Removed spaces.

File: Assignment 3-JS/script/script.js (hunk 32 above)
Comment (ayushiagrawal2000) [reply Assignment 3-JS/script/script.js]: Removed extra line breaks.

File: Assignment 3-JS/script/script.js (hunk 33 above)
Comment (ayushiagrawal2000) [reply Assignment 3-JS/script/script.js]: Changed it to monthDays and also changed other variable names.

File: Assignment 3-JS/template/index.html (hunk 34 above)
Comment (ayushiagrawal2000) [reply Assignment 3-JS/template/index.html]: Fixed it.

**Pull request merged**